frontend/src/components/NewGroup.tsx

```diff
@@ -64,37 +79,40 @@ function NewGroup(props: { onClose: () => void; session: Session | null }) {
   ];
   const [isCreatingGroup, setIsCreatingGroup] = React.useState<boolean>(false); // new state
   const [selectedTab, setSelectedTab] = React.useState<number>(0); // new state
-  const [user, setUser] = useState(null); // new state
-
-  useEffect(() => {
-    const fetchUserData = async () => {
-      try {
-        // Fetch user data from the server using the API
-        const response = await fetch(`${API_URL}/user`, {
-          method: "GET",
-          headers: {
-            "Content-Type": "application/json",
-            access_token: props.session!.access_token,
-          },
-        });
-
-        if (!response.ok) {
-          throw new Error("Network response was not ok");
-        }
 
-        const userData = await response.json();
-        //using the auth user email we create a new Friend(group member) then add the new friend into the friend state
-        const myself = new Friend(null, null, userData.user.email, true);
-        setFriends([myself]);
-        setUser(userData.user.email);
-      } catch (error) {
-        console.error("There was a problem fetching user data:", error);
-      }
-    };
+  let pricePerMember = 0;
+  if (splitMode === "equally" && selectedSubscription && _friends.length > 0) {
+    pricePerMember = selectedSubscription.cost / _friends.length;
+  }
+  const [customAmounts, setCustomAmounts] = useState<Record<string, number>>(
```

Comment:
don't need a separate state if this info can be inside of Friend's subscription_cost instead.